src/freenet/client/async/SplitFileInserterStorage.java

```diff
@@ -218,7 +223,16 @@
 
     private final int overallStatusLength;
     
+    /** Protects cooldown calculations and noBlocksToSend. You MUST NOT take this lock while 
+     * holding any other locks!!!
```

Comment:
See my branch refactor-one-getter-per-segment. But it only does requests so far, not inserts, needs re-cherry-picking against official next and needs cleaning up (and is relatively large).